server/celery-queue/tasks.py

```diff
@@ -32,20 +56,66 @@ def solve(domain_url: str, problem_url: str, solver: str) -> str:
 
     problem_file = f'{tmpfolder}/{os.path.basename(problem_url)}'
     download_file(problem_url, problem_file)
-
-    res = subprocess.run(f'{solver} {domain_file} {problem_file}',
-                            stdout=subprocess.PIPE, stderr=subprocess.PIPE,
-                            executable='/bin/bash', encoding='utf-8',
-                            shell=True, cwd=tmpfolder)
+    
+    # Will generate a single output file (the plan) which is returned via HTTP
+    command = f"{solver} {domain_file} {problem_file}"
+    res = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                        executable='/bin/bash', encoding='utf-8',
+                        shell=True, cwd=tmpfolder)
 
     # remove the tmp/fies once we finish
-    os.remove( domain_file )
-    os.remove( problem_file )
+    os.remove(domain_file)
+    os.remove(problem_file)
+    
+    plan = retrieve_output_file(PACKAGES[solver]['endpoint']['return']['file'], tmpfolder)
+    
     shutil.rmtree(tmpfolder)
 
-    return {'stdout': res.stdout, 'stderr': res.stderr}
+    return {'stdout': res.stdout, 'stderr': res.stderr, 'plan':plan}
+
+# Running generic planutils packages with no solver-specific assumptions
+@celery.task(name='tasks.run.package')
+def run_package(package: str, arguments:dict, call:str, output_file:str):
+    tmpfolder = tempfile.mkdtemp()
+    # Write files and replace args in the call string
+    for k, v in arguments.items():
+        if v['type'] == 'file':
+            # Need to write to a temp file
+            path_to_file = write_to_temp_file(k, v['value'], tmpfolder)
+            # k is a file, we want to replace with the file path
+            call = call.replace("{%s}" % k, k)
+        else:
+            # k needs to be replaced with the value 
+            call = call.replace("{%s}" % k, str(v['value']))
+
+    # Run the command
+    res = subprocess.run(call, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                        executable='/bin/bash', encoding='utf-8',
+                        shell=True, cwd=tmpfolder)
+    
+    output = retrieve_output_file(output_file, tmpfolder)
+
+    return {"stdout":res.stdout, "stderr":res.stderr, "call":call, "output":output}
+    
+# Running a generic solver with string PDDL inputs 
+@celery.task(name='tasks.solve.string')
```

Comment:
Just like the suggestion above, can you fold this into the previous by just having the right details/defaults in advance?

Reply:
Going to get this done today

Reply:
Do you think that we should combine tasks.solve and tasks.solve.string, with the separate task for running generic packages - or should I look to combine all 3 into tasks.run.package?

@haz

Reply:
Ultimately it should be just the single package handling. We'll want to handle the legacy `/solve` endpoint for a while, but just repackage the data into a call for the `tasks.run.package`

Reply:
Sounds good